### PR TITLE
[v8.0]Propagate more env variable in the SingularityCE

### PR DESCRIPTION
**Pull request by chaen**
Without this, CERN jobs started failing upload (could not find the CA, still investigating), and RAL job with oldish `ROOT` versions fail (redirector setup)

BEGINRELEASENOTES

*Resources
FIX: Propagate environment variables from host to container.

ENDRELEASENOTES

### CODE REVIEW

File: src/DIRAC/Resources/Computing/SingularityComputingElement.py [hunk 1]
```diff
@@ -85,6 +86,18 @@
 """
 
 
+ENV_VAR_WHITELIST = [
+    r"TERM",
+    r"VOMS_.*",
+    r"X509_.*",
+    r"XRD_.*",
+    r"Xrd.*",
+    r"DIRAC.*",  # will take also DIRAC and DIRACOS
```
Comment (chrisburr) [inline src/DIRAC/Resources/Computing/SingularityComputingElement.py]: This is probably an issue

**Comment by sfayer** [comment]
Hi,

We're having related problems: In no-install mode, the DIRACSYSCONFIG variable containing the pilot.cfg location doesn't get pulled through to the innner container so none of the command line tools work... It sounds like this would fix that (assuming pilot.cfg is actually somewhere in the bit that's mounted in the container, I haven't checked that yet).

However, I suspect some of the variables will be wrong if __installDIRACInContainer is True as the paths outside & inside the container may not match up, so perhaps it should only happen if not __installDIRACInContainer (or maybe different sets of vars depending on the mode)?

Regards,
Simon

File: src/DIRAC/Resources/Computing/SingularityComputingElement.py (hunk 1 above)
Comment (chaen) [reply src/DIRAC/Resources/Computing/SingularityComputingElement.py]: then we can take `DIRAC_`

**Comment by chaen** [comment]
> However, I suspect some of the variables will be wrong if __installDIRACInContainer is True as the paths outside & inside the container may not match up, so perhaps it should only happen if not __installDIRACInContainer (or maybe different sets of vars depending on the mode)?

I think that makes sense

**Comment by chrisburr** [comment]
> perhaps it should only happen if not __installDIRACInContainer (or maybe different sets of vars depending on the mode)?

Perhaps, though I wonder if we can drop the `__installDIRACInContainer` option?

**Comment by sfayer** [comment]
> > perhaps it should only happen if not __installDIRACInContainer (or maybe different sets of vars depending on the mode)?
> 
> Perhaps, though I wonder if we can drop the `__installDIRACInContainer` option?

We were planning on switching over, but as I say, we ran into pilot.cfg issues and are back running in install mode for now: If we can fix those then I don't see it being a problem.

**Comment by chaen** [comment]
How about treating this `pilot.cfg` like `X509_USER_PROXY` and copy it in the work area ?

**Comment by sfayer** [comment]
> How about treating this `pilot.cfg` like `X509_USER_PROXY` and copy it in the work area ?

Yes, I think that would work.

**Comment by chaen** [comment]
> > How about treating this `pilot.cfg` like `X509_USER_PROXY` and copy it in the work area ?
> 
> Yes, I think that would work.

In fact, isn't it what is already done here  ? 
https://github.com/DIRACGrid/DIRAC/blob/3ab57e91cc4de7841551e79619aef1bb7c55b51e/src/DIRAC/Resources/Computing/SingularityComputingElement.py#L261-L285

**Comment by sfayer** [comment]
> > > How about treating this `pilot.cfg` like `X509_USER_PROXY` and copy it in the work area ?
> > 
> > 
> > Yes, I think that would work.
> 
> In fact, isn't it what is already done here ?
> 
> https://github.com/DIRACGrid/DIRAC/blob/3ab57e91cc4de7841551e79619aef1bb7c55b51e/src/DIRAC/Resources/Computing/SingularityComputingElement.py#L261-L285

It looks that way... In which case it just needs the environment variable setting correctly inside the container to point at the copy...

**Comment by chaen** [comment]
You guys had so many trouble with this `pilot.cfg` not being found so I am not even sure which one is the correct one :laughing: 
Setting `DIRACSYSCONFIG` to the `pilot.cfg` would solve the issue ?

**Comment by sfayer** [comment]
> You guys had so many trouble with this `pilot.cfg` not being found so I am not even sure which one is the correct one laughing Setting `DIRACSYSCONFIG` to the `pilot.cfg` would solve the issue ?


Yes please... DIRACSYSCONFIG should be set to the full "inside the container" path to pilot.cfg.

**Comment by chaen** [comment]
Does that seem reasonable to everybody ?

**Comment by sfayer** [comment]
Looks good to me.

**Comment by chaen** [comment]
we will try it in the LHCb prod environment

**Review by chrisburr — APPROVED** [review]

**Comment by chaen** [comment]
It works just fine in LHCb. @sfayer @marianne013 did you have a chance at trying it ?

**Comment by marianne013** [comment]
We ran it on the preprod server using the tag directly from your repo. It worked, otherwise you would have heard about it :-D.  Whether we covered all use cases remains to be seen.

**Comment by chaen** [comment]
:laughing: OK, merging then

**Pull request merged**